akp/types/configmap.go

```diff
@@ -13,6 +13,44 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func ToFilteredConfigMapTFModel(ctx context.Context, diagnostics *diag.Diagnostics, data *structpb.Struct, oldCM tftypes.Map) tftypes.Map {
+	if data == nil || len(data.AsMap()) == 0 {
+		if !oldCM.IsUnknown() && (oldCM.IsNull() || len(oldCM.Elements()) == 0) {
+			return oldCM
+		}
+	}
+
+	oldMap := make(map[string]interface{})
```

Comment:
```suggestion
	oldMap := make(map[string]interface{}, len(oldCM))
```
or similar, fewer alloc calls

Reply:
done, thanks!